community/cloud-foundation/templates/gke/gke.py

```diff
@@ -0,0 +1,138 @@
+# Copyright 2018 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" This template creates a Google Kubernetes Engine cluster. """
+
+
+def generate_config(context):
+    """ Entry point for the deployment resources. """
+
+    resources = []
+    outputs = []
+    project_id = context.env['project']
+    properties = context.properties
+    cluster_type = properties.get('clusterLocationType')
+    propc = properties['cluster']
+    name = propc.get('name') or context.env['name']
+    gke_cluster = {
+        'name': name,
+        'type': '',
+        'properties':
+            {
+                'cluster':
+                    {
+                        'name':
+                            name + '-cluster',
+                        'initialNodeCount':
+                            propc.get('initialNodeCount'),
+                        'initialClusterVersion':
+                            propc.get('initialClusterVersion')
+                    }
+            }
+    }
+
+    if cluster_type == 'Regional':
+        provider = 'gcp-types/container-v1beta1:projects.locations.clusters'
+        if not properties.get('region'):
+            raise KeyError(
+                "region is a required property for a {} Cluster."
+                .format(cluster_type)
+            )
+        parent = 'projects/{}/locations/{}'.format(
+            project_id,
+            properties.get('region')
+        )
+        gke_cluster['properties']['parent'] = parent
+
+    elif cluster_type == 'Zonal':
+        provider = 'container.v1.cluster'
+        if not properties.get('zone'):
+            raise KeyError(
+                "zone is a required property for a {} Cluster."
+                .format(cluster_type)
+            )
+        gke_cluster['properties']['zone'] = properties.get('zone')
+
+    gke_cluster['type'] = provider
```

Comment:
I see, thanks for pointing that out - I missed it; in summary it will always one of the conditional blocks since we have a default clusterLocationType defined in the schema